app/helpers/Emails/NotificationsHelper/Comments/SolutionCommentsEmailsSender.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Helpers\Notifications;
+
+use App\Helpers\EmailHelper;
+use App\Model\Entity\AssignmentSolution;
+use App\Model\Entity\Comment;
+use App\Model\Entity\ReferenceExerciseSolution;
+use Latte;
+use Nette\Utils\Arrays;
+
+/**
+ * Sending emails on solution commentary.
+ */
+class SolutionCommentsEmailsSender {
+
+  /** @var EmailHelper */
+  private $emailHelper;
+  /** @var string */
+  private $sender;
+  /** @var string */
+  private $assignmentSolutionCommentPrefix;
+  /** @var string */
+  private $referenceSolutionCommentPrefix;
+
+  /**
+   * Constructor.
+   * @param EmailHelper $emailHelper
+   * @param array $params
+   */
+  public function __construct(EmailHelper $emailHelper, array $params) {
+    $this->emailHelper = $emailHelper;
+    $this->sender = Arrays::get($params, ["emails", "from"], "noreply@recodex.cz");
+    $this->assignmentSolutionCommentPrefix = Arrays::get($params, ["emails", "assignmentSolutionCommentPrefix"], "Assignment Solution Comment - ");
+    $this->referenceSolutionCommentPrefix = Arrays::get($params, ["emails", "referenceSolutionCommentPrefix"], "Reference Solution Comment - ");
+  }
+
+  /**
+   * Internal solution send comment method.
+   * @param AssignmentSolution|ReferenceExerciseSolution $solution
+   * @param Comment $comment
+   * @return bool
+   */
+  private function sendSolutionComment($solution, Comment $comment): bool {
+    if ($comment->isPrivate()) {
+      // comment was private, therefore do not send email to others
+      return true;
+    }
+
+    $baseSolution = $solution->getSolution();
+    $subject = $this->assignmentSolutionCommentPrefix . $baseSolution->getAuthor()->getName();
+    $recipients = [$baseSolution->getAuthor()->getEmail()];
+    foreach ($comment->getThread()->findAllPublic() as $pComment) {
+      $user = $pComment->getUser();
+      if (!$user->getSettings()->getSolutionCommentsEmails()) {
+        continue;
+      }
+      $recipients[] = $user->getEmail();
+    }
+
+    // make user emails unique, so the emails won't be multiplied
+    $recipients = array_unique($recipients);
+
+    // filter out the author of the comment, it is pointless to send email to that user
+    $recipients = array_filter($recipients, function (string $email) use ($comment) {
+      return $email !== $comment->getUser()->getEmail();
+    });
```

Comment:
I know that premature optimizations are sqrt() of all evil; however, in this case it would be much nicer to user emails as keys in $recepients array. The uniqueness will take care of itself and removing logged user's email would not require filtering the array. And the only drawback is that you need to use array_keys($recepients) in the send() method...